src/measure_theory/bochner_integration.lean

```diff
@@ -199,18 +199,99 @@ and prove basic property of this integral.
 open finset
 
 variables [normed_group E] [measurable_space E] [normed_group F]
-variables {μ : measure α}
-
-/-- For simple functions with a `normed_group` as codomain, being integrable is the same as having
-    finite volume support. -/
-lemma integrable_iff_fin_meas_supp {f : α →ₛ E} {μ : measure α} :
-  integrable f μ ↔ f.fin_meas_supp μ :=
-calc integrable f μ ↔ ∫⁻ x, f.map (coe ∘ nnnorm : E → ℝ≥0∞) x ∂μ < ∞ :
-  and_iff_right f.ae_measurable
-... ↔ (f.map (coe ∘ nnnorm : E → ℝ≥0∞)).lintegral μ < ∞ : by rw lintegral_eq_lintegral
-... ↔ (f.map (coe ∘ nnnorm : E → ℝ≥0∞)).fin_meas_supp μ : iff.symm $
-  fin_meas_supp.iff_lintegral_lt_top $ eventually_of_forall $ λ x, coe_lt_top
-... ↔ _ : fin_meas_supp.map_iff $ λ b, coe_eq_zero.trans nnnorm_eq_zero
+variables {μ : measure α} {p : ℝ≥0∞}
+
+/-!
+#### Properties of simple functions
+
+A simple function `f : α →ₛ E` into a normed group `E` verifies, for a measure `μ`:
+- `mem_ℒp f 0 μ` and `mem_ℒp f ∞ μ`, since `f` is a.e.-measurable and bounded,
+- for `0 < p < ∞`, `mem_ℒp f p μ ↔ integrable f μ ↔ f.fin_meas_supp μ ↔ ∀ y ≠ 0, μ (f ⁻¹' {y}) < ∞`.
+-/
+
+lemma exists_forall_norm_le (f : α →ₛ F) : ∃ C, ∀ x, ∥f x∥ ≤ C :=
+exists_forall_le (f.map (λ x, ∥x∥))
+
+lemma mem_ℒp_zero (f : α →ₛ E) (μ : measure α) : mem_ℒp f 0 μ :=
+mem_ℒp_zero_iff_ae_measurable.mpr f.ae_measurable
+
+lemma mem_ℒp_top (f : α →ₛ E) (μ : measure α) : mem_ℒp f ∞ μ :=
+begin
+  obtain ⟨C, hfC⟩ := f.exists_forall_norm_le,
+  exact mem_ℒp_top_of_bound f.ae_measurable C (eventually_of_forall hfC),
+end
```

Comment:
```suggestion
let ⟨C, hfC⟩ := f.exists_forall_norm_le in
mem_ℒp_top_of_bound f.ae_measurable C $ eventually_of_forall hfC
```

Reply:
That one and the other suggestions I have not yet incorporated into my code look like they only remove the  `begin...end` (by replacing an `obtain` by a `let` or other minor change). I will eventually commit your changes since less lines is better but I am curious: is there another gain? Are those proofs faster to elaborate for example, or is it only about the two lines? (right now I can't remember how I would test the speed)

Reply:
To be honest I'm not entirely sure, I just didn't see a reason for them to be in tactic mode when they could be in term mode.

Reply:
For a two-lines proof I agree! As soon as the proof is a bit longer, I have a strong preference for tactic mode, which I find more readable. Anyway, let's commit those changes!